java/src/jmri/jmrix/xpa/XpaTurnout.java

```diff
@@ -36,62 +29,47 @@
     public XpaTurnout(int number, XpaSystemConnectionMemo m) {
         super(m.getSystemPrefix() + "T" + number);
         _number = number;
+        _prefix = m.getSystemPrefix();
         tc = m.getXpaTrafficController();
     }
 
     public int getNumber() {
         return _number;
     }
 
-    // Handle a request to change state by sending a formatted DCC packet
+    /**
+     * {@inheritDoc}
+     */
     @Override
-    synchronized protected void forwardCommandChangeToLayout(int s) {
+    synchronized protected void forwardCommandChangeToLayout(int newState) {
         XpaMessage m;
         // sort out states
-        if ((s & _mClosed) != 0) {
-            // first look for the double case, which we can't handle
-            if ((s & _mThrown ) != 0) {
-                // this is the disaster case!
-                log.error("Cannot command both CLOSED and THROWN " + s);
-                return;
+        if ((newState & Turnout.CLOSED) != 0) {
+            if (statesOk(newState)) {
+                // send a CLOSED command (or THROWN if inverted)
+                m = XpaMessage.getSwitchMsg(_number, getInverted());
             } else {
-                // send a CLOSED command
-                m = XpaMessage.getSwitchNormalMsg(_number);
+                log.error("Cannot command both CLOSED and THROWN {}", newState);
+                return;
             }
         } else {
-            // send a THROWN command
-            m = XpaMessage.getSwitchReverseMsg(_number);
+            // send a THROWN command (or CLOSED if inverted)
+            m = XpaMessage.getSwitchMsg(_number, !getInverted());
         }
         tc.sendXpaMessage(m, null);
     }
 
     @Override
     protected void turnoutPushbuttonLockout(boolean _pushButtonLockout) {
-        if (log.isDebugEnabled()) {
-            log.debug("Send command to " + (_pushButtonLockout ? "Lock" : "Unlock") + " Pushbutton PT" + _number);
-        }
-    }
-
-    @Override
-    synchronized public void setInverted(boolean inverted) {
-        log.debug("Inverting Turnout State for turnout {}", getSystemName() );
-        _inverted = inverted;
-        if (inverted) {
-            _mThrown = Turnout.CLOSED;
-            _mClosed = Turnout.THROWN;
-        } else {
-            _mThrown = Turnout.THROWN;
-            _mClosed = Turnout.CLOSED;
-        }
-        super.setInverted(inverted);
+        log.debug("Send command to {} Pushbutton {}T{}", (_pushButtonLockout ? "Lock" : "Unlock"),
+                _prefix, _number);
```

Comment:
You should be able to replace the {}T{} with {} and just use the system name as the only parameter.

Reply:
done (and removed unused _prefix)